### PR TITLE
Update refresh button to use API

### PR DESCRIPTION
**Pull request by bgrgicak**
### Changes proposed in this Pull Request:

The refresh button works by redirecting to the old subscriptions page and back. We need to replace this with API calls before we can remove the old subscriptions page.

This PR implements a new button that uses the `/refresh` API endpoint to upload the subscriptions context with latest data from Woo.com.

The new button also has a busy state to indicate that it's working.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the new subscriptions page and connect to Woo.com if you are not connected
2. Click the refresh button
3. The button should be set to busy
4. Confirm that it sent an API request to `/wc/v3/marketplace/refresh`
5. When the request finishes, the button should stop spinning and a notice will appear

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Subscriptions refresh button uses the API instead of redirect.

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx [hunk 1]
```diff
@@ -89,6 +77,9 @@ export default function MySubscriptions(): JSX.Element {
 
 	return (
 		<div className="woocommerce-marketplace__my-subscriptions">
+			<section className="woocommerce-marketplace__my-subscriptions__notices">
```
Comment (bgrgicak) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.tsx:80]: This was removed from the feature branch, so I'm adding it back.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss [hunk 2]
```diff
@@ -6,9 +6,19 @@
...
+
+	&.is-busy {
+		@keyframes spin {
+			0% { transform: rotate(0deg); }
+			100% { transform: rotate(360deg); }
+		  }
+		.woocommerce-marketplace__refresh-subscriptions__icon {
+			animation: spin 2s linear infinite;
+		}
+	}
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss:28]: Can we please add this on line 24 to address one of Poli's comments in 18757-gh-Automattic/woocommerce.com?

> Refresh button:
>  Is that the right spacing between the icon and text? Is this something coming from Woo core? There should be spacing between the icon and text, 4px.

```scss
.woocommerce-marketplace__refresh-subscriptions__icon {
	margin-right: $grid-unit-05;
}
```

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/refresh-button.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	return (
+		<Button
+			className="woocommerce-marketplace__refresh-subscriptions"
+			onClick={ refresh }
+			isBusy={ isLoading }
+		>
+			<img
+				src={ RefreshIcon }
+				alt={ __( 'Refresh subscriptions', 'woocommerce' ) }
+				className="woocommerce-marketplace__refresh-subscriptions__icon"
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/refresh-button.tsx]: We're stretching the `block__element--modifier` format rather with these classnames. I know we already have `woocommerce-marketplace__my-subscriptions__installed` and so on, but I would try and at least limit the mutations by using `woocommerce-marketplace__refresh-subscriptions-icon` here.

**Review by andfinally — APPROVED** [review]
Beautiful change, thank you! 🙌 I think this is safe to merge with one review.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/refresh-button.tsx (hunk 3 above)
Comment (bgrgicak) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/refresh-button.tsx]: Thanks, you are correct this wouldn't be valid BEM.

**Review by mlaetitia — APPROVED** [review]
Works great!

**Pull request merged**